src/rust/lib/frp/src/nodes.rs

```diff
@@ -303,6 +303,16 @@ impl Network {
     }
 
 
+    // === Filter ===
+
+    /// Passes exactly those incoming events that satisfy the predicate `p`.
+    pub fn filter<T:EventOutput>
+    (&self, label:Label, src:&T, p:impl'static+Fn(&Output<T>)->bool)
```

Comment:
would `p:'static + impl Fn(&Output<T>)->bool` work here as well? If so, please use it as it's more clear what it does. In fact I haven't ever seen `impl 'static` usage anywhere else, but I'm glad it works!

Reply:
This would not work. But if you find that clearer then I can use an explicit type parameter as in
```
pub fn filter<T:EventOutput, P:'static+Fn(&Output<T>)->bool>
(&self, label:Label, src:&T, p:P)
```

Reply:
I'm ok with the current style, however, the most clear to me (personal preference) in case of such a complex type, would be this style, as it provides a very short and clean look at the strucutre of the signature before getting into details about the constraints (which we use across this lib):

```rust
pub fn filter<T,F>(&self, label:Label, src:&T, f:F) -> Stream<Output<T>>
where T:EventOutput, F:'static+Fn(&Output<T>)->bool {
    ...
}
```

Btw, you are using this style below in filter_map - so lets use it here as well :)